### ISSUE TITLE
Items queried (not loaded) cannot be modified

### ISSUE THREAD

**Issue by phillip-haydon** [JasperFx/marten#117]
So the problem I have is that I have some setup data, which looks at existing data in the database and changes it if it doesn't match.

i.e ImageSettings, in the Database a Thumbnail is set to Quality = 80%

So I load ALL settings `session.Query<ImageSettings>().ToList()`

Then iterate over them, and modify them.

```
var existingSettings = session.Query<ImageSettings>().EmptyIfNull().ToList();

// Update or delete existing settings
foreach (var imgSetting in existingSettings) 
{
    var setting = ImgSettings.FirstOrDefault(x => x.ImageType == imgSetting.ImageType);

    if (setting != null)
    {
        imgSetting.Height = setting.Height;
        imgSetting.Width = setting.Width;
        imgSetting.Quality = setting.Quality;
        imgSetting.Auto = setting.Auto;
        imgSetting.NameFormat = setting.NameFormat;
        imgSetting.Crop = setting.Crop;
        imgSetting.Usage = setting.Usage;
    }
    else
    {
        session.Delete(imgSetting);
    }
}
```

These changes wont get tracked and persisted. 

The temporary solution to this problem is re-load each one and save it in the loop.

```
// Update or delete existing settings
foreach (var imgSetting in existingSettings)
{
    var toSave = session.Load<ImageSettings>(imgSetting.Id);
    var setting = ImgSettings.FirstOrDefault(x => x.ImageType == imgSetting.ImageType);

    if (setting != null)
    {
        toSave.Height = setting.Height;
        toSave.Width = setting.Width;
        toSave.Quality = setting.Quality;
        toSave.Auto = setting.Auto;
        toSave.NameFormat = setting.NameFormat;
        toSave.Crop = setting.Crop;
        toSave.Usage = setting.Usage;
    }
    else
    {
        session.Delete(toSave);
    }

    session.SaveChanges();
}
```

**Comment by jeremydmiller** [comment]
@phillip-haydon Couldn't you just put an `IDocumentSession.Store()` call in the first loop to tell Marten that the doc needs to be persisted? You don't have to depend on the automatic dirty tracking.

**Comment by phillip-haydon** [comment]
Funnily enough it works, but I'm not sure if the API is intuitive having to call store for each item you want to modify. Possibly just need to update docs for this.

**Comment by jeremydmiller** [comment]
That behavior *is* documented. How did you create your `IDocumentSession`? You have to opt into the heavy automatic dirty checking (because RavenDb leaked memory like you wouldn't believe if you used their equivalent to this naively)

**Comment by phillip-haydon** [comment]
```
using (var session = store.OpenSession())
{

}
```

I was just opening a normal session, rather than a lightweight one mentioned in the docs. 

http://jasperfx.github.io/marten/documentation/documents/persisting/

This is what I read last night, but coming from RavenDB I (poorly) assumed that if I used a normal session, that the changes would be tracked and persisted. 

I didn't realise I need to call `.Store(...)` in all scenarios that don't use a dirty session. 

However I did try a dirty session and my changes were not persisted either :(


OK so there's no issue, just my bad understanding :)
